pyrealm/phenology/fapar_limitation.py

```diff
@@ -0,0 +1,415 @@
+        x: Array of values to be converted to annual values. Should be either daily (
+            same datetimes as growing_season) or subdaily (same datetimes as datetimes
+            array)
+        datetimes: Datetimes of the measurements as np.datetime64 arrays.
+        growing_season: Bool array of days, indicating whether they are ain growing
+            season or not.
+        method: Either "total" (sum all values of the year) or "mean" (take the mean
+            of all values of the year)
+    """
+
+    # Extract years from datetimes
+    all_years = datetimes.astype("datetime64[Y]")
+
+    if len(x) == len(growing_season):  # this is daily data
+        daily_x = x
+        n_days = len(x)
+        years_by_day = all_years.view()
+        obs_per_day = int(len(datetimes) / n_days)
+        years_by_day.shape = tuple([n_days, obs_per_day, *list(all_years.shape[1:])])
+    elif len(x) == len(datetimes):  # this is subdaily data
+        # Create scaler object to handle conversion between scales
+        scaler = AcclimationModel(datetimes)
+        scaler.set_nearest(np.timedelta64(12, "h"))
+        # Convert values to daily to match with growing_season
+        daily_x = scaler.get_daily_means(x)
+        years_by_day = scaler.get_window_values(np.asarray(all_years))
+    else:
+        raise ValueError("Input array does not fit datetimes nor growing_season array")
+
+    # Which years are present?
+    years = np.unique(all_years)
+
+    # Compute annual totals or means, only taking into account the days which are in
+    # growing season.
+
+    annual_x = np.zeros(len(years))
+
+    if method == "total":
+        for i in range(len(years)):
+            annual_x[i] = sum(
+                daily_x[growing_season & (years_by_day[:, 0] == years[i])]
+            )
+    elif method == "mean":
+        for i in range(len(years)):
+            annual_x[i] = np.mean(
+                daily_x[growing_season & (years_by_day[:, 0] == years[i])]
+            )
+    else:
+        raise ValueError("No valid method given for annual values")
+
+    return annual_x
+
+
+def daily_to_subdaily(
+    x: NDArray,
+    datetimes: NDArray[np.datetime64],
+) -> NDArray:
+    """Broadcasts an array of the entity x from daily values to subdaily values.
+
+    Args:
+        x: Array of daily values.
+        datetimes: Subdaily datetimes as np.datetime64 array.
+    """
+
+    n_days = len(x)
+    obs_per_day = int(len(datetimes) / n_days)
+
+    subdaily_x = np.repeat(x, obs_per_day)
+
+    return subdaily_x
+
+
+def convert_precipitation_to_molar(precip_mm: NDArray) -> NDArray:
+    """Convert precipitation from mm/m2 to mol/m2.
+
+    - 1 mm/m2 = 1000000 mm3 = 1000 mL = 1 L
+    - Molar mass of water = 18g
+    - Assuming density = 1 (but really temp varying), molar volume = 18mL
+    - So 1 mm/m2 = 1000 / 18 ~ 55.55 mols/m2
+    """
+
+    water_mm_to_mol = 1000 / 18
+    precip_molar = precip_mm * water_mm_to_mol
+
+    return precip_molar
```

Comment:
I think this can go? There is now a temperature and pressure sensitive version in `core.water`